### PR TITLE
fix: avoid crash on empty NFT details

### LINKED ISSUE #7397: [Task]: Fix empty NFT freezing the app
### Task description

[Example NFT in the explorer](https://explorer.shimmer.network/testnet/addr/rms1zzk7v0tfswwsn385nfcnznd0xgwd0ks48ktv9e3w5z6j5jxpvyzk58sczmj)

image_720

### Requirements

Looking at the details should not freeze/crash the app.

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by VmMad**
## Summary

Fixes #7397 
> Bring back missing optional chaining to avoid crashes.

...

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [x] Linux
    -   [ ] Windows

### Instructions

> Mint empty nft with the CLI. Click on the NFT details. It should not crash.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [x] I have followed the contribution guidelines for this project
-   [x] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/wallet/utils/getBech32AddressFromAddressTypes.ts [hunk 1]
```diff
@@ -2,9 +2,9 @@ import { api } from '@core/profile-manager'
 import { Address, AddressType, AliasAddress, Ed25519Address, NftAddress } from '@iota/sdk/out/types'
 import { getNetworkHrp } from '@core/profile'
 
-export function getBech32AddressFromAddressTypes(address: Address): string | undefined {
+export function getBech32AddressFromAddressTypes(address: Address | undefined): string | undefined {
```
Comment (cpl121) [inline packages/shared/lib/core/wallet/utils/getBech32AddressFromAddressTypes.ts:5]: Is it necessary to set this `undefined` here?
I think that `?` is enough

Comment (marc2332) [reply packages/shared/lib/core/wallet/utils/getBech32AddressFromAddressTypes.ts:5]: The fact that he is using `?` means that the address might be undefined at some point, so it's better that the argument type reflects this case

**Review by cpl121 — APPROVED** [review]

File: packages/shared/lib/core/wallet/utils/getBech32AddressFromAddressTypes.ts (hunk 1 above)
Comment (begonaalvarezd) [inline packages/shared/lib/core/wallet/utils/getBech32AddressFromAddressTypes.ts:5]: I think the bugfix for this issue should not come from making this argument optional, as it must be compulsory that you pass an address in order to be able to retrieve its bech32, what needs fixing is all the type errors we have in the code where we are actually calling this function with something that might be undefined, eg
image
We need to make sure in the code that we are not calling this with something that is undefined, and only call it when the types are right

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
The crash is still here 🙏🏼 
image

**Review by begonaalvarezd — APPROVED** [review]
🚀 now it works

**Pull request merged**